toolchain/check/testdata/var/fail_namespace_conflict.carbon

```diff
@@ -0,0 +1,54 @@
+// Part of the Carbon Language project, under the Apache License v2.0 with LLVM
+// Exceptions. See /LICENSE for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+// AUTOUPDATE
+
+namespace A;
+
+// CHECK:STDERR: fail_namespace_conflict.carbon:[[@LINE+9]]:5: ERROR: Duplicate name being declared in the same scope.
+// CHECK:STDERR: var A: b;
+// CHECK:STDERR:     ^
+// CHECK:STDERR: fail_namespace_conflict.carbon:[[@LINE-5]]:1: Name is previously declared here.
+// CHECK:STDERR: namespace A;
+// CHECK:STDERR: ^~~~~~~~~~~~
+// CHECK:STDERR: fail_namespace_conflict.carbon:[[@LINE+3]]:8: ERROR: Name `b` not found.
+// CHECK:STDERR: var A: b;
+// CHECK:STDERR:        ^
+var A: b;
```

Comment:
I'd prefer something that avoids an error on the type so its clear what this is trying to test. The type name lookup errors feel like noise in this context.

Happy for you to fix this here and submit w/o another round, or do this in a follow-up.

(Here and below as well of course.)

Reply:
Done, verified it's not necessary for the crash.